be/src/storage/lake/compaction_scheduler.h

```diff
@@ -196,16 +201,19 @@ class CompactionScheduler {
 
     void steal_task(int start_index, std::unique_ptr<CompactionTaskContext>* context);
 
-    int choose_task_queue_by_txn_id(int64_t txn_id) const { return txn_id % _task_queue_count; }
+    int choose_task_queue_by_txn_id(int64_t txn_id) const { return txn_id % _task_queue_count.load(); }
 
     bool txn_log_exists(int64_t tablet_id, int64_t txn_id) const;
 
     TabletManager* _tablet_mgr;
     Limiter _limiter;
     bthread::Mutex _contexts_lock;
     butil::LinkedList<CompactionTaskContext> _contexts;
-    int _task_queue_count;
-    TaskQueue* _task_queues;
+
+    std::atomic<int> _task_queue_count;
+    // use a mutex to protect the resize operation of _task_queues at runtime
+    std::mutex _task_queues_mutex;
+    std::vector<TaskQueue> _task_queues;
```

Comment:
How about wrapping these fields together in a utility class that provides queue-like APIs?